libcloudforensics/providers/gcp/internal/gke.py

```diff
@@ -36,24 +41,105 @@ def GkeApi(self) -> 'googleapiclient.discovery.Resource':
     Returns:
       googleapiclient.discovery.Resource: A Google Container service object.
     """
+    return common.CreateService('container', self.GKE_API_VERSION)
 
-    return common.CreateService(
-        'container', self.GKE_API_VERSION)
 
-  def GetCluster(self, name: str) -> Dict[str, Any]:
-    """ Gets the details of a specific cluster.
+class GkeCluster(GoogleKubernetesEngine):
```

Comment:
why not have a GetK8sCluster here to get an authenticated K8sCluster object from GkeCluster object, the caller then can do:
```
gke_cluster = GkeCluster('project-id', 'zone-id', 'cluster-id')
k8s_cluster = gke_cluster.GetK8sCluster()
```
instead of
```
gke_cluster = GkeCluster('project-id', 'zone-id', 'cluster-id')
api_client = gke_cluster.GetCredentials()
k8s_cluster = K8sCluster(api_client)
```